sdk/swap-service/tests/oneinch.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { ChainFamilyMap, ChainInfo } from '@summerfi/sdk-common/chains'
+import { Address, Percentage, Token, TokenAmount } from '@summerfi/sdk-common/common'
+import { getSwapManager, SwapData, SwapProviderType } from '~swap-service'
+
+describe('OneInch | SwapManager | Integration', () => {
+  it('should provide swap data', async () => {
+    // SwapManager
+    const swapManager = getSwapManager()
+
+    // ChainInfo
+    const chainInfo: ChainInfo = ChainFamilyMap.Ethereum.Mainnet
+
+    // Tokens
+    const WETH = await Token.createFrom({
+      chainInfo,
+      address: Address.createFrom({ hexValue: '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2' }),
+      symbol: 'WETH',
+      name: 'Wrapped Ether',
+      decimals: 18,
+    })
+
+    const DAI = await Token.createFrom({
+      chainInfo,
+      address: Address.createFrom({ hexValue: '0x6B175474E89094C44Da98b954EedeAC495271d0F' }),
+      symbol: 'DAI',
+      name: 'Dai Stablecoin',
+      decimals: 18,
+    })
+
+    // TokenAmount
+    const fromAmount: TokenAmount = TokenAmount.createFrom({
+      token: WETH,
+      amount: '34.5',
+    })
+
+    const toMinimumAmount: TokenAmount = TokenAmount.createFrom({
+      token: DAI,
+      amount: '1000.0',
+    })
+
+    // Address
+    const recipient: Address = Address.createFrom({
+      hexValue: '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266',
+    })
+
+    // Percentage
+    const slippage: Percentage = Percentage.createFrom({ percentage: 2.0 })
```

Comment:
I'm leaning towards creating a Slippage class extending Percentage, would be much more clear to use in implementation. what you think? any arguments againts that?

Reply:
Or is it so clear that there will be no confusion whatsoever?

Reply:
I'm actually wondering from the client side POV

Reply:
Mmm, not sure about it. A slippage is a percentage, but I'm not sure that it deserves its own class. I'm not against though, just trying to understand what would be less confusing, as you say

Reply:
No big deal, I agree might not deserve separate class at this point as it doesn't seem that important.